lib/model/compose.dart

```diff
@@ -199,3 +199,35 @@ String mention(User user, {bool silent = false}) {
 String inlineLink(String visibleText, Uri? destination) {
   return '[$visibleText](${destination?.toString() ?? ''})';
 }
+
+/// What we show while fetching the target message's raw Markdown.
+String quoteAndReplyPlaceholder(PerAccountStore store, {
+  required Message message
+}) {
+  final sender = store.users[message.senderId];
+  assert(sender != null);
+  final url = narrowLink(store, SendableNarrow.ofMessage(message, selfUserId: store.account.userId));
+  final senderPart = '${mention(sender!, silent: true)} ${inlineLink('said', url)}: '; // TODO(i18n) ?
+  return '$senderPart*(loading message ${message.id})*\n'; // TODO(i18n) ?
+}
+
+/// Quote-and-reply syntax.
+///
+/// The result looks like it does in Zulip web:
+///
+///     @_**Iago|5** [said](link to message):
+///     ```quote
+///     message content
+///     ```
+String quoteAndReply(PerAccountStore store, {
+  required Message message,
+  required String rawContent,
+}) {
+  final sender = store.users[message.senderId];
+  assert(sender != null);
+  final url = narrowLink(store,
+    SendableNarrow.ofMessage(message, selfUserId: store.account.userId),
+    nearMessageId: message.id);
+  final senderLine = '${mention(sender!, silent: true)} ${inlineLink('said', url)}:\n'; // TODO(i18n) ?
+  return senderLine + wrapWithBacktickFence(content: rawContent, infoString: 'quote');
+}
```

Comment:
Could write tests for these, I suppose 😅—but they mostly just use code that already has test coverage.

Reply:
I think my question above about having the message ID in the link perhaps answers this :slightly_smiling_face:.

Because the various pieces that these functions put together already have their own test coverage, these might not need more than a single test case each. But that single test case would provide coverage of how these functions do put those pieces together, like the presence of `/near/12345` in the link and the use of silent mentions.

Reply:
Ah yeah, makes sense. For the "expected" string, I guess we have a choice to make it a more-or-less literal string, or put that together using `inlineLink` and friends since they're covered already.

I've gone the more-or-less literal string route for my next revision but would happily switch to the other.